src/LndClient/Data/LndEnv.hs

```diff
@@ -1,58 +1,133 @@
 {-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE OverloadedStrings #-}
 
 module LndClient.Data.LndEnv
   ( LndEnv (..),
+    RawConfig (..),
     LndWalletPassword (..),
     LndTlsCert (..),
     LndHexMacaroon (..),
     LndHost (..),
     LndPort (..),
     newLndEnv,
     readLndEnv,
+    createLndTlsCert,
+    createLndPort,
   )
 where
 
+import Data.Aeson as A ((.:), FromJSON (..), Value (..))
 import Data.ByteString.Char8 as C8
 import qualified Data.PEM as Pem
+import Data.Scientific
+import Data.Text as T (unpack)
 import Data.Text.Lazy as LT
 import Data.X509
 import Env
 import LndClient.Class
 import LndClient.Data.Type
-import LndClient.Import.External
+import LndClient.Import.External as Ex
 import LndClient.Util as U
 import Network.GRPC.HighLevel.Generated
 import Network.GRPC.LowLevel.Client
+import Text.Read
 
-newtype LndWalletPassword = LndWalletPassword ByteString
-  deriving (PersistField, PersistFieldSql, Eq)
+newtype LndWalletPassword = LndWalletPassword Text
+  deriving (PersistField, PersistFieldSql, Eq, FromJSON, IsString, Show)
```

Comment:
Oh man, please be careful with `Show` type class. We don't want any type of credentials or sensitive personal data to be able to be shown! It's really bad idea to make `LndWalletPassword ` and similar types showable. As future refactoring step we need to remove `Show` class from pretty much everywhere in our codebase.

I understand that you needed it for Hspec test, but let's do it properly. It's not easy question how to do it. There are so many ways to do it, this question deserves separate discussion.. We can implement `Show` instead of deriving it, showing for example harmless hash of data instead of data itself, or just show something like `(LndWalletPassword "SECRET"). We can create our own type class `ShowDebug` which is different from `Show` and use it in tests only and don't export from library in any form. We can create separate newtypes for test purposes. So many possibilities.
 
As shortest solution for now, we can refactor Hspec test a bit, something like 

```haskell
myRawEnv == fromJSON myRawEnvJSON `shouldBe` True
```

This will not give you nice output which parts are not matching if test fails, but there it's probably fine because so simple test will never fail. This refactoring will require to derive `Eq` which is really harmless if compare with `Show`

Reply:
Btw `Eq` is already derived there

Reply:
Simplified test, actually we don't need to check actual fields, but overall successfull parsing. So updated test, removed Show deriving and remove some constructors from LndEnv module export.

Reply:
For more complex cases actually I think this is the case where orphan instances might be useful. We define (derive) orphan instances of Show in our test folder, and export in tests where it's needed. Minimal effort, but we still have nice output in tests and don't expose Show instances from library